substrate-node/pallets/pallet-smart-contract/src/types.rs

```diff
@@ -47,24 +63,36 @@ impl<T: Config> Contract<T> {
 
     pub fn get_node_id(&self) -> u32 {
         match self.contract_type.clone() {
-            ContractData::RentContract(c) => c.node_id,
-            ContractData::NodeContract(c) => c.node_id,
+            ContractData::CapacityReservationContract(c) => c.node_id,
+            ContractData::DeploymentContract(_) => 0,
             ContractData::NameContract(_) => 0,
         }
     }
 }
 
-#[derive(Clone, Eq, PartialEq, RuntimeDebugNoBound, Encode, Decode, TypeInfo, MaxEncodedLen)]
+#[derive(Clone, Eq, PartialEq, RuntimeDebugNoBound, Encode, Decode, TypeInfo)]
 #[scale_info(skip_type_params(T))]
 #[codec(mel_bound())]
-pub struct NodeContract<T: Config> {
+pub struct CapacityReservationContract {
     pub node_id: u32,
+    pub resources: ConsumableResources,
+    pub group_id: Option<u32>,
+    pub public_ips: u32,
+    pub deployment_contracts: Vec<u64>,
```

Comment:
Why are we saving the deployment contract ids on the capacity reservation contract?

Reply:
In case node_deleted is called. Then we need to delete all contracts. I just wanted to avoid having to create a new storage map.

Reply:
Wouldn't this be cleaned up by the billing? I think it's checked somewhere if the node exists, if not the contract is removed.

Reply:
You mean on line 1497? Yes indeed but that means we won't be freeing the resources right away... So I would have to add code to free the resources there. Plus freeing will then only happen at the next billing of that contract.